src/openrct2/object/SmallSceneryObject.cpp

```diff
@@ -115,22 +115,22 @@ void SmallSceneryObject::DrawPreview(rct_drawpixelinfo* dpi, int32_t width, int3
         screenCoords.y -= 12;
     }
 
-    gfx_draw_sprite(dpi, imageId, screenCoords, 0);
+    gfx_draw_sprite(dpi, imageId, screenCoords);
 
     if (_legacyType.HasFlag(SMALL_SCENERY_FLAG_HAS_GLASS))
     {
-        imageId = (_legacyType.image + 4) | IMAGE_TYPE_TRANSPARENT | (EnumValue(GlassPaletteIds[COLOUR_BORDEAUX_RED]) << 19);
-        gfx_draw_sprite(dpi, imageId, screenCoords, 0);
+        imageId = ImageId(_legacyType.image + 4, EnumValue(GlassPaletteIds[COLOUR_BORDEAUX_RED])).WithBlended(true);
```

Comment:
You can't pass a palette via the constructor, you must use `WithRemap(GlassPaletteIds[COLOUR_BORDEAUX_RED])`

I am however still getting the crash with what I am working on, still investigating.